homeassistant/components/reolink/host.py

```diff
@@ -359,6 +366,34 @@ def unregister_webhook(self):
         webhook.async_unregister(self._hass, self.webhook_id)
         self.webhook_id = None
 
+    async def _poll_all_motion(self, *_) -> None:
+        """Poll motion and AI states until the first ONVIF push is received."""
+        if self._webhook_reachable.is_set():
+            # ONVIF push is working, stop polling
+            self._cancel_poll = None
+            return
+
+        try:
+            await self._api.get_motion_state_all_ch()
+        except (
+            aiohttp.ClientConnectorError,
+            asyncio.TimeoutError,
+            ReolinkError,
+        ) as err:
+            _LOGGER.error(
+                "Reolink error while polling motion state for host %s:%s: %s",
+                self._api.host,
+                self._api.port,
+                str(err),
+            )
+
+        async_dispatcher_send(self._hass, f"{self.webhook_id}_all", {})
```

Comment:
Because that will only run `async_write_ha_state` on all binary sensors.
I added the following comment: 
```
        # After receving the new motion states in the upstream lib,
        # update the binary sensors with async_write_ha_state
        # The same dispatch as for the webhook can be used
```